src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -559,6 +559,8 @@ void FFN<
   // Reset the network parameters with the given initialization rule.
   NetworkInitialization<InitializationRuleType> networkInit(initializeRule);
   networkInit.Initialize(network.Network(), parameters);
+  // Override the weight matrix if necessary.
+  network.CustomInitialize(parameters, network.WeightSize(), 1);
```

Comment:
I think we could actually make the API simpler and remove all the parameters here if we reworked the `FFN` class a little bit to call `CustomInitialize` *after* `SetLayerMemory` is called.  Then, each layer can simply use its own internal memory (an alias of `parameters`).  It's up to you if you want to make that change, but at the very least I think it would be a good thing to remove the unused `cols` parameter, and rename the `rows` parameter to `elements` (or something), since it simply represents the number of elements that the layer holds.  In fact, since `MultiLayer` uses aliases, you could even remove the `elements`/`rows` parameters, since it is implicit in the matrix parameter anyway.